logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt

```diff
@@ -122,19 +132,41 @@ internal class SlowSyncManager(
         }
     }
 
-    private suspend fun handleCriteriaResolution(syncCriteriaResolution: SyncCriteriaResolution, isSlowSyncNeeded: Boolean) {
+    private suspend fun handleCriteriaResolution(syncCriteriaResolution: SyncCriteriaResolution, isSlowSyncNeeded: SlowSyncParam) {
         if (syncCriteriaResolution is SyncCriteriaResolution.Ready) {
             // START SYNC IF NEEDED
             logger.i("SlowSync criteria ready, checking if SlowSync is needed or already performed")
-            if (isSlowSyncNeeded) {
-                logger.i("Starting SlowSync as all criteria are met and it wasn't performed recently")
-                performSlowSync()
-                logger.i("SlowSync completed. Updating last completion instant")
-                slowSyncRepository.setSlowSyncVersion(CURRENT_VERSION)
-                slowSyncRepository.setLastSlowSyncCompletionInstant(DateTimeUtil.currentInstant())
-            } else {
-                logger.i("No need to perform SlowSync. Marking as Complete")
+
+            when (isSlowSyncNeeded) {
+                SlowSyncParam.LastSlowSyncTooOld,
+                SlowSyncParam.NotPerformedBefore -> {
+                    logger.i("Starting SlowSync as all criteria are met and it wasn't performed recently")
+                    performSlowSync(emptyList())
+                    logger.i("SlowSync completed. Updating last completion instant")
+                    slowSyncRepository.setSlowSyncVersion(CURRENT_VERSION)
+                    slowSyncRepository.setLastSlowSyncCompletionInstant(DateTimeUtil.currentInstant())
```

Comment:
suggestion: all the code around `performSlowSync` could be moved to inside of it.

```suggestion
                    performSlowSync(migrationSteps = emptyList())
```